test/splitter.js

```diff
@@ -0,0 +1,206 @@
+
+        await truffleAssert.reverts(
+            splitter.splitDeposit(recipientTwo, recipientOne,{from: sender, value: 4}),
+            "Pausable: paused"
+        );
+        checkEventNotEmitted();
+    });
+
+    it("SplitDeposit is unpausable", async () => {
+
+        await splitter.pause({from: contractOwnerAddress});
+        await splitter.unpause({from: contractOwnerAddress});
+
+        const txObj = await splitter.splitDeposit(recipientTwo, recipientOne,{from: sender, value: 5});
+
+        truffleAssert.eventEmitted(txObj, 'Deposit');
+    })
+
+
+    it('Second recipient can successfully withdraw 2 wei', async () => {
+
+        const depositAmount = toBN(5);
+        const withDrawAmount = toBN(2);
+        
+        // Deposit into the contract
+        await splitter.splitDeposit(recipientOne, recipientTwo, {from: sender, value: depositAmount});
+
+        // Take a snapshot of the second recipient's ETH balance
+        const initRecipientTwoEthBalance = toBN(await web3.eth.getBalance(recipientTwo));
+
+        // Recipient withdraws
+        const txObj = await splitter.withdraw(withDrawAmount, {from: recipientTwo});
+        const cost = await getGasCost(txObj);
+
+        // Get the recipient's new ETH and contract balances
+        const recipientTwoEthBalance = toBN(await web3.eth.getBalance(recipientTwo));
+        const recipientTwoOwed = toBN(await splitter.balances(recipientTwo));
+
+        // Calculate the expected new ETH and contract balances
+        const expectedRecipientTwoEthBalance = initRecipientTwoEthBalance.sub(cost).add(withDrawAmount).toString(10);
+        const expectedRecipientTwoOwed = "0";
+
+        assert.strictEqual(recipientTwoEthBalance.toString(10), expectedRecipientTwoEthBalance);
+        assert.strictEqual(recipientTwoOwed.toString(10), expectedRecipientTwoOwed);
+
+        truffleAssert.eventEmitted(txObj, "WithDraw", (ev) => {
+            return  ev.withdrawer === recipientTwo &&
+                    ev.amount.toString(10) === "2";
+        }, 'TestEvent should be emitted with correct parameters');
+
+    });
+
+    it("Second recipient attempts to withdraw more than is available in the balance", async () => {
+
+        withDrawAmount = toWei(toBN(5), "ether");
+
+        await truffleAssert.reverts(
+            splitter.withdraw(withDrawAmount, {from: recipientTwo}),
+            "There are insufficient funds"
+        );
+        checkEventNotEmitted();
+    });
+
+    it("Second recipient attempts to withdraw zero", async () => {
+
+        await truffleAssert.reverts(
+            splitter.withdraw(toBN(0), {from: recipientTwo}),
+            "The value must be greater than 0"
+        );
+        checkEventNotEmitted();
+    });
+
+
+    it("Withdraw is pausable", async () => {
+
+        await splitter.pause({from: contractOwnerAddress});
+
+        await truffleAssert.reverts(
+            splitter.withdraw(toBN(0), {from: recipientTwo}),
```

Comment:
A value that we do not expect to be wrong, like `"2"` would make more sense because as it stands the withdraw cannot happen because:

1. there is 0 in the contract
2. there is 0 owed
3. the contract is paused
4. it is asking to withdraw 0

That does not make for a very meaningful test where the stated intent is to test 3 in isolation.

Reply:
Understood